client.go

```diff
@@ -428,6 +440,19 @@ func (c *Client) handleMessage(msgType int, data []byte) error {
 				return err
 			}
 			c.sigChainBlockHash = sigChainBlockHash
+
+		// client auth
+		case authChallengeAction:
+			var challenge string
+			if err := json.Unmarshal(*msg["Challenge"], &challenge); err != nil {
+				return err
+			}
+			byteChallenge, _ := hex.DecodeString(challenge)
+			signature, err := crypto.Sign(c.account.PrivateKey, byteChallenge)
+			if err != nil {
+				return err
+			}
+			c.chChallengeSignature <- signature
```

Comment:
Need to handle channel full (node somehow sent multiple challenge) and avoid blocking

Reply:
fixed